weave-js/src/components/Panel2/PanelComp.tsx

```diff
@@ -186,6 +186,13 @@ export function useUpdateConfig2<C>(props: {
   updateConfig2?(change: (oldConfig: C) => Partial<C>): void;
 }) {
   const {config, updateConfig} = props;
+
+  // By using a ref, we can ensure that downstream components can call
+  // updateConfig2 multiple times between re-renders, and the config will
+  // always be the latest.
+
+  const workingConfig = useRef(config);
+  workingConfig.current = config;
```

Comment:
What if the props change -- the resulting config value won't be synced to the workingConfig right?

Reply:
Ahhh i typed out a response and closed the tab :( ... ok shorter version:
- Basically workingConfig.current will always be at least as fresh as props.config.
- This resource is helpful: https://react.dev/learn/queueing-a-series-of-state-updates

The render loop goes:
- props.config comes in
- `const workingConfig = useRef(config);`
- `workingConfig.current = config;`
- ...
- User triggers callback that has updateConfig2 calls
-- `workingConfig.current` updated with new version + updateConfig called 
--- depending on the config provider, `updateConfig` could trigger a setState or a redux update
- Next render loop:
-- props.config has most-recent state, loop continues.

Now, if the config is updated by another process during this loop, it is possible that one of the updates is agnostic to the others, resulting in a non-applied update. But I think this is a general problem with any non-callback based state update